informer/integration/test/collector/collector_test.go

```diff
@@ -3,204 +3,94 @@
 package collector
 
 import (
-	"fmt"
+	"context"
 	"testing"
 	"time"
 
-	"github.com/giantswarm/microerror"
-	"github.com/giantswarm/operatorkit/informer"
+	"github.com/giantswarm/operatorkit/informer/collector"
 	"github.com/prometheus/client_golang/prometheus"
-	prommodel "github.com/prometheus/client_model/go"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
 )
 
-const (
-	grace int64 = 500
-)
-
-var (
-	configMapName = "test-configmap"
-	configmap     = &corev1.ConfigMap{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "v1",
-			Kind:       "ConfigMap",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      configMapName,
-			Namespace: namespace,
-		},
-		Data: map[string]string{},
-	}
-
-	kindLabel      = "kind"
-	nameLabel      = "name"
-	namespaceLabel = "namespace"
-	emptyValue     = ""
-)
-
-// Test_Informer_Collector_Basic is an integration test for basic collector operations.
-// The test verifies the metrics emitted by the collector.
-func Test_Informer_Collector_Basic(t *testing.T) {
-	testCases := []struct {
-		name string
-
-		// Events to run before starting the collector.
-		setupEvents []func(kubernetes.Interface) error
-		// Events to run just after starting the collector.
-		events []func(kubernetes.Interface) error
-
-		// Descriptions of the metrics we expect to be emitted.
-		expectedMetrics []*prometheus.Desc
-	}{
-		{
-			name: "Test one create and one update event emits only one creation_timestamp metric",
-
-			setupEvents: []func(kubernetes.Interface) error{
-				// Create the configmap to force a create event.
-				func(k8sClient kubernetes.Interface) error {
-					if _, err := k8sClient.CoreV1().ConfigMaps(namespace).Create(configmap); err != nil {
-						return microerror.Mask(err)
-					}
-
-					return nil
-				},
-			},
-
-			events: []func(kubernetes.Interface) error{
-				// Update the configmap to force an update event,
-				// within the 1 second of the collector watching for events.
-				func(k8sClient kubernetes.Interface) error {
-					configmap.Data["foo"] = "bar"
-					if _, err := k8sClient.CoreV1().ConfigMaps(namespace).Update(configmap); err != nil {
-						return microerror.Mask(err)
+func Test_Informer_Collector_MultipleEvents(t *testing.T) {
```

Comment:
This thing got completely rewritten. Maybe look at the plain file since the diff is odd. The test is way simpler now IMO and covers the case.